rasa_core/training/interactive.py

```diff
@@ -561,21 +571,51 @@ def _split_conversation_at_restarts(evts):
     return sub_conversations
 
 
-def _write_stories_to_file(export_file_path, sender_id, endpoint):
+def _collect_messages(evts):
+    # type: (List[Dict[Text, Any]]) -> List[Dict[Text, Any]]
+    """Collect the message text and parsed data from the UserMessage events into a list"""
+
+    msgs = []
+
+    for evt in evts:
+        if evt.get("event") == "user":
+            data = evt.get("parse_data")
+            msg = Message.build(data["text"], data["intent"]["name"], data["entities"])
+            msgs.append(msg)
+
+    return msgs
+
+
+def _write_to_file(export_file_paths, sender_id, endpoint):
     # type: (Text, Text, EndpointConfig) -> None
-    """Write the conversation of the sender_id to the file path."""
+    """Write the conversation and nlu data of the sender_id to the file paths."""
 
     tracker = retrieve_tracker(endpoint, sender_id)
     evts = tracker.get("events", [])
 
     sub_conversations = _split_conversation_at_restarts(evts)
+    msgs = _collect_messages(evts)
+    previous_examples = TrainingData()
+    failsafe_toggle = 'w'
 
-    with io.open(export_file_path, 'a', encoding="utf-8") as f:
+    try:
+        previous_examples = load_data(export_file_paths[1])
+    except:
+        failsafe_toggle = 'a'
+        logger.exception("Could not load NLU data from existing file, new data will be appended to prevent loss of"
```

Comment:
but this will create an invalid file right? I think in that case it would make more sense to e.g. store the data into a temp file instead of the existing one and let the user handle this